### PR TITLE
[Modules] Redis cache module

### PR DESCRIPTION
**Pull request by itpropro**
# Description

Added redis cache module with the following ResourceModule provided defaults:

- Telemetry
- Resource locks
- Diagnostics settings
- RBAC

# Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (readme)
- [X] I did format my code

### CODE REVIEW

File: arm/Microsoft.Cache/redis/.test/deploy.full.test.bicep [hunk 1]
```diff
@@ -0,0 +1,31 @@
+var location = 'westeurope'
```
Comment (MrMCake) [inline arm/Microsoft.Cache/redis/.test/deploy.full.test.bicep:1]: Note: The CI environment does not yet support Bicep test files.

Instead the tests expect a `.parameters` folder with JSON-Parameter files with a naming convention `*.parameters.json`.

File: arm/Microsoft.Cache/redis/deploy.bicep [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+resource redisCache_lock 'Microsoft.Authorization/locks@2017-04-01' = if (!empty(lock)) {
+  name: '${redisCache.name}-${lock}-lock'
+  properties: {
+    level: any(lock)
+    notes: lock == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+  }
+  scope: redisCache
+}
+
+resource redisCache_diagnosticSettingName 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = if (!empty(diagnosticStorageAccountId) || !empty(diagnosticWorkspaceId) || !empty(diagnosticEventHubAuthorizationRuleId) || !empty(diagnosticEventHubName)) {
```
Comment (MrMCake) [inline arm/Microsoft.Cache/redis/deploy.bicep:219]:
```suggestion
resource redisCache_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = if (!empty(diagnosticStorageAccountId) || !empty(diagnosticWorkspaceId) || !empty(diagnosticEventHubAuthorizationRuleId) || !empty(diagnosticEventHubName)) {
```

File: arm/Microsoft.Cache/redis/deploy.bicep [hunk 3]
```diff
@@ -0,0 +1,259 @@
+@description('Optional. The location to deploy the Redis cache service.')
```
Comment (MrMCake) [inline arm/Microsoft.Cache/redis/deploy.bicep:1]: Please also add the missing GitHub & AzureDevOps pipelines.

File: arm/Microsoft.Cache/redis/.test/deploy.full.test.bicep (hunk 1 above)
Comment (itpropro) [reply arm/Microsoft.Cache/redis/.test/deploy.full.test.bicep]: Added

**Comment by itpropro** [comment]
Added all the missing parts @MrMCake

**Comment by MrMCake** [comment]
@itpropro, thanks for that. I'll migrate the PR later today :)

**Pull request merged**